pkg/apis/componentconfig/types_pluginargs.go

```diff
@@ -78,3 +79,32 @@ type RemoveDuplicatesArgs struct {
 	Namespaces        *api.Namespaces
 	ExcludeOwnerKinds []string
 }
+
+type Percentage float64
+type ResourceThresholds map[v1.ResourceName]Percentage
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type LowNodeUtilizationArgs struct {
+	metav1.TypeMeta
+
+	UseDeviationThresholds     bool
+	Thresholds                 ResourceThresholds
+	TargetThresholds           ResourceThresholds
+	ThresholdPriority          int32
+	ThresholdPriorityClassName string
+	NumberOfNodes              int
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type HighNodeUtilizationArgs struct {
+	metav1.TypeMeta
+
+	UseDeviationThresholds     bool
+	Thresholds                 ResourceThresholds
+	TargetThresholds           ResourceThresholds
```

Comment:
hi. please correct me if i am wrong, in my opinion, because `HighNodeUtilization ` and `LowNodeUtilization ` too many duplicates, so if i remove `TargetThresholds ` from  `HighNodeUtilization`, then like function https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/nodeutilization/highnodeutilization.go#L55 you can not pass a nil to here, that's why i keep it as before.

Reply:
targetThresholds is set at https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/strategies/nodeutilization/highnodeutilization.go#L48 again. When `TargetThresholds` gets removed from `HighNodeUtilizationArgs`, it can be set directly. I.e. from
```go
func (h *HighNodeUtilization) Balance(ctx context.Context, nodes []*v1.Node) *framework.Status {
	thresholds := h.args.Thresholds
	targetThresholds := h.args.TargetThresholds

	targetThresholds = make(componentconfig.ResourceThresholds)

	setDefaultForThresholds(thresholds, targetThresholds)
	resourceNames := getResourceNames(targetThresholds)

	sourceNodes, highNodes := classifyNodes(
		getNodeUsage(nodes, resourceNames, h.handle.GetPodsAssignedToNodeFunc()),
		getNodeThresholds(nodes, thresholds, targetThresholds, resourceNames, h.handle.GetPodsAssignedToNodeFunc(), false),
```
to
```go
func (h *HighNodeUtilization) Balance(ctx context.Context, nodes []*v1.Node) *framework.Status {
	thresholds := h.args.Thresholds
	targetThresholds := make(componentconfig.ResourceThresholds)

	setDefaultForThresholds(thresholds, targetThresholds)
	resourceNames := getResourceNames(targetThresholds)

	sourceNodes, highNodes := classifyNodes(
		getNodeUsage(nodes, resourceNames, h.handle.GetPodsAssignedToNodeFunc()),
		getNodeThresholds(nodes, thresholds, targetThresholds, resourceNames, h.handle.GetPodsAssignedToNodeFunc(), false),
```
I.e. removing
```
targetThresholds := h.args.TargetThresholds
targetThresholds = make(componentconfig.ResourceThresholds)
```
and replacing with
```
targetThresholds := make(componentconfig.ResourceThresholds)
```

Reply:
get it. i will update soon